source/robot_model/src/Model.cpp

```diff
@@ -268,18 +243,18 @@ Eigen::MatrixXd Model::cwln_weighted_matrix(const state_representation::JointPos
   for (int n = 0; n < this->robot_model_.nq; ++n) {
     double d = 1;
     W_b(n, n) = 1;
-    if (joint_positions.get_positions()[n] < this->robot_model_.lowerPositionLimit[n] + margin) {
-      if (joint_positions.get_positions()[n] < this->robot_model_.lowerPositionLimit[n]) {
+    if (joint_positions.data()[n] < this->robot_model_.lowerPositionLimit[n] + margin) {
```

Comment:
can you briefly explain why you've changed this? Is it because you think the `get_positions()` method is mainly intended for the base class JointState, so that data() should be the default "getter" for the derived classes?

Reply:
yes exactly. I think the `derived.get_derived()` is redundant. I was actually thinking of removing it altogether for derived classes to only have the `data()` as default getter. But I figured we can also keep it to be consistent with the base class and also because there is not `set_data` that would be default setter.